proposals/0156-high-level-interface-foundation.md

```diff
@@ -34,387 +37,248 @@ The high level interface sits on top of the management layer. It should provide
-| Not connected | FULL          | `appDidConnect:`<br>`appDidFinishLaunch:`<br>`appDidBecomeActive:`    | Media app resumed after ignition cycle and became visible on HMI      |
-| NONE          | FULL          | `appDidFinishLaunch:`<br>`appDidBecomeActive:`                        | App was selected by the user (or system)                              |
-| BACKGROUND    | FULL          | `appDidBecomeActive:`                                                 | Background app was selected by the user                               |
-| LIMITED       | FULL          | `appDidBecomeActive:`                                                 | Active media app was selected by the user                             |
-
-### View Controller lifecycle
----
+/**
+ * The central class coordinating the SDL application running on the SDL enabled system
+ * By default every application contains a single instance of SDLApplication. 
+ */
+@interface SDLApplication
 
-Based on the MVC pattern used by UIKit the high level interface should introduce the SDL view controller in order to manage and control views. App developers need to subclass the base view controller class and implement the use cases. As view controllers are stackable the app developer can separate and structure the SDL related code.
+/** Returns the singleton app instance. */
+@property (strong, nonatomic, class, readonly) SDLApplication *sharedApplication;
 
-A view controller can exist in a set of states. Below flow shows the possible transitions between loading view, view to appear or to disappear.
+/** The delegate of the SDL application. */
+@property (weak, nonatomic) id<SDLApplicationDelegate> delegate;
 
-![view-controller-lifecycle](../assets/proposals/0156-high-level-interface-foundation/view-controller-lifecycle.png)
+/** The current state of the application. */
+@property (assign, nonatomic, readonly) SDLApplicationState applicationState;
 
-![view-controller-overview](../assets/proposals/0156-high-level-interface-foundation/view-controller-overview.png)
+/** Describes whether or not streaming audio of the media application is currently audible through the system. */
+@property (assign, nonatomic, readonly) SDLAudioStreamingState audibleState;
 
-#### `Init`
+/** Indicates whether or not a user-initiated interaction is in progress, and if so, in what mode (i.e. MENU or VR). */
+@property (assign, nonatomic, readonly) SDLSystemContext systemContext;
 
-View controllers are instantiated when they are pushed to the view controller manager. In the case of the root view controller it would be instantiated once a connection is established to the head unit (app is registered). In this state there is no view loaded. The app developer should not load data or perform other memory/time intense code at this time.
+/** The configuration the SDL application. (bypass to manager) */
```

Comment:
I think this should be either "The configuration of the SDL application." or "The SDL application configuration."